cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -78,6 +78,10 @@ type node struct {
 	// tenant define the list of objects which are tenant for the node, no matter if the node has a direct OwnerReference to the object or if
 	// the node is linked to a object indirectly in the OwnerReference chain.
 	tenant map[*node]empty
+
+	// rawObject holds the object that is referenced when creating the node.
+	// the object can then be referenced latter in a save / restore cycle
+	rawObject *unstructured.Unstructured
```

Comment:
Might be I'm missing something, but I don't think this field is necessary.
It appears that it is set during restore and the used only in getDiscoveryFileTypes, which is called before restore...

Reply:
The `rawObject` field is used in the restoreTargetObject method.

Since these objects are being restored from file, we need the original, raw object from the object file originating from the backed up management cluster. In my method, I set the object graph to have references to these original objects since we can't populate the object with the typical `get` method using a client (like we do in the move sequence).

Without this field, I'm not sure how we would accurately restore from file.

Reply:
Now that getDiscoveryFileTypes has been removed, should we remove this field?

Reply:
I believe this field is still essential. 

Without it, we would have no actual object reference when restoring objects. It is used [here](https://github.com/kubernetes-sigs/cluster-api/pull/4808/files#diff-3458bf9e9c245fe801573500c33de535f05d3a5887f080a405b819e3c0fed415R864) as part of applying an object to a target cluster during restore. In the `move` method, we can generate some object key and query the source management cluster to fill in the rest of the object. During a restore, we cannot rely on a source management cluster to populate objects. So, without a source management cluster to interrogate, we have to rely on the raw object given from the file and populated when the object was added to the graph.

Reply:
Ok, got your point, however I think we should avoid storing the objects always when this is necessary only for restore.
What about:
- renaming this field to restoredObject
- introduce a addrestoredObj function that calls addObj and set restoredObject, and use this new func in restore
- remove proposed changes in `objToNode` and in `ownerToVirtualNode` so we avoid allocating objects in all the other operations

Reply:
This is now done